sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/AvgSqlAggregator.java

```diff
@@ -115,9 +115,13 @@ public Aggregation toDruidAggregation(
           project,
           Iterables.getOnlyElement(aggregateCall.getArgList())
       );
-      String vc = virtualColumnRegistry.getVirtualColumnByExpression(arg, resolutionArg.getType());
-      fieldName = vc != null ? vc : null;
-      expression = vc != null ? null : arg.getExpression();
+      if (arg.getType() == DruidExpression.NodeType.LITERAL) {
```

Comment:
This doesn't really feel like somethign that each implementation should have to do to me...  Could we not like pass in the virtualColumnRegistry to the DruidExpression and ask it to tell us the expression to use?  Where, like, in the literal case, it would give us an expression that's just a literal and in the non-literal case it would register the thing and then return the name to use for it?

Reply:
i did consolidate this into a static method in `SimpleSqlAggregator` while you were typing this comment, I haven't really decided on baking it into `VirtualColumnRegistry` in some way yet though, will think about it some more.

Reply:
after further thought i've decided to just always use a virtual column, even for literals, so this code is simplified a bit now